ibis/tests/expr/test_signature.py

```diff
@@ -198,32 +217,77 @@ class NoHooves(Farm):
     assert g1.goats == 0
 
 
+def test_copy_default():
+    default = []
+
+    class Op(Annotable):
+        arg = rlz.optional(rlz.instance_of(list), default=default)
+
+    op = Op()
+    assert op.arg is not default
+
+
 def test_slots_are_inherited_and_overridable():
     class Op(Annotable):
         __slots__ = ('_cache',)  # first definition
         arg = Argument(lambda x: x)
 
     class StringOp(Op):
-        arg = Argument(str)  # inherit
+        arg = Argument(str)  # new overridden slot
```

Comment:
We do redefine the `arg` slot here, although we could use the parent slot too, but then we need to aggregate the parent slots to check whether the slot already exists or not.

Reply:
I don't know if this is reasonably possible. What would happen in the case of `C(A, B)` where `A(Annotable)` and `B(Annotable)` and `A` and `B` have a slot with the same name?

Reply:
That's explicitly disallowed by python, see https://github.com/ibis-project/ibis/pull/3105/files#diff-21051dad17424e3006baa46b3c0555827349335eb54ea9d6eb290cda9992c13fR276-R280